torch/_inductor/codegen/common.py

```diff
@@ -48,8 +48,8 @@ def index_prevent_reordering(index: typing.List[sympy.Expr], index_vars, sizes):
 class OpDtypeClassifier:
```

Comment:
This class is primarily used in `deduce_node_dtype`. Is it simpler to directly check the ops inside `deduce_node_dtype` without this class?